components/automate-cli/cmd/chef-automate/automateClusterCtlUtil.go

```diff
@@ -137,22 +146,268 @@ func bootstrapEnv(dm deployManager) error {
+	manifestPath := ""
+	var airgapMetadata airgap.UnpackMetadata
+	if offlineMode {
+		writer.Title("Installing artifact")
+		metadata, err := airgap.Unpack(deployCmdFlags.airgap)
+		if err != nil {
+			return status.Annotate(err, status.AirgapUnpackInstallBundleError)
+		}
+		airgapMetadata = metadata
+		manifestPath = api.AirgapManifestPath
+
+		// We need to set the path for the hab binary so that the deployer does not
+		// try to go to the internet to get it
+		pathEnv := os.Getenv("PATH")
+
+		err = os.Setenv("PATH", fmt.Sprintf("%s:%s", path.Dir(metadata.HabBinPath), pathEnv))
+		if err != nil {
+			return err
+		}
+	} else {
+		manifestPath = conf.Deployment.GetV1().GetSvc().GetManifestDirectory().GetValue()
+	}
+
 	manifestProvider := manifest.NewLocalHartManifestProvider(
-		mc.NewDefaultClient(conf.Deployment.GetV1().GetSvc().GetManifestDirectory().GetValue()),
+		mc.NewDefaultClient(manifestPath),
 		conf.Deployment.GetV1().GetSvc().GetHartifactsPath().GetValue(),
 		conf.Deployment.GetV1().GetSvc().GetOverrideOrigin().GetValue())
-	offlineMode := deployCmdFlags.airgap != ""
+
 	err := client.DeployHA(writer, conf, manifestProvider, version.BuildTime, offlineMode)
 	if err != nil && !status.IsStatusError(err) {
 		return status.Annotate(err, status.DeployError)
 	}
+	if offlineMode {
+		err := moveAirgapToTransferDir(airgapMetadata)
+		if err != nil {
+			return status.Annotate(err, status.DeployError)
+		}
+	}
 	err = dm.generateConfig()
 	if err != nil {
 		return status.Annotate(err, status.DeployError)
 	}
 	return nil
 }
 
+/*
+In airgap deployement of A2HA we need to have two AIB files in /hab/a2_deploy_workspace/terraform/transfer_files/
+first airgap file will automate generated airgap bundle,
+second file will be backend airgap bundle which is same as automate(frontend) airgap bundle but without header.
+also we need to keep MD5 hash files both frontend and backend bundles.
+we also need to have following files in /hab/a2_deploy_workspace/terraform dir
+1. a2ha_aib_fe.auto.tfvars --> it will contain info of frontend airgap bundle
+2. a2ha_aib_be.auto.tfvars --> it will contain info of backend airgap bundle
+1. a2ha_manifest.auto.tfvars --> it will contain info of required packages to deploy a2ha.
+*/
+func moveAirgapToTransferDir(airgapMetadata airgap.UnpackMetadata) error {
+	if len(deployCmdFlags.airgap) > 0 {
+		var bundleName string = filepath.Base(deployCmdFlags.airgap)
+		if strings.Contains(bundleName, "automate") {
+			bundleName = strings.ReplaceAll(bundleName, "automate", "frontend")
+		}
+		err := copyFileContents(deployCmdFlags.airgap, (AIRGAP_HA_TRANS_DIR_PATH + bundleName))
+		if err != nil {
+			return err
+		}
+		//generating md5 sum for frontend bundle
+		err = generateChecksumFile(AIRGAP_HA_TRANS_DIR_PATH+bundleName, AIRGAP_HA_TRANS_DIR_PATH+bundleName+".md5")
+		if err != nil {
+			return err
+		}
+		//generating frontend auto tfvars
+		var frontendAutotfvarsTemplate string = `
```

Comment:
move the constant string on top of the file, and then replace {bundleName} here while you use it further.